### PR TITLE
Port & Implement new pools page

### LINKED ISSUE #207: Port & Implement new Pools page
* Move any store/data logic to store package
* Add new features/UI in figma
* Use new UI components

Pools page component should be just wiring store + UI components together instead of business logic.

[Figma](https://www.figma.com/file/Wyb0yUw8vGSW1AUAYutt7L/Revised-Pools-(Draft%3A-APR-tooltip%2C-visual-hierarchy-changes)?node-id=250%3A40124)

### PR DESCRIPTION
**Pull request by delivan**
Closes #207

I'm struggling to match some types of table component. Please check if I'm using it as you intended. @jonator

### CODE REVIEW

File: packages/web/stores/imperator-queries/pools.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  last_update_at: number;
+  data: PoolMetric[];
+}> {
+  constructor(kvStore: KVStore) {
+    super(kvStore, "/fees/v1/pools");
+
+    makeObservable(this);
+  }
+
+  readonly makePoolsWithMetric = computedFn(
```
Comment (jonator) [inline packages/web/stores/imperator-queries/pools.ts:57]: Why does this accept a list of pools? Given that it is memoized I think it would be more general to calculate for one pool at a time. Call it `makePoolFeeMetrics` then it returns `ObservablePoolWithFeeMetrics` object.

File: packages/web/stores/imperator-queries/pools.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import { makeObservable, observable, action, computed } from "mobx";
+import { computedFn } from "mobx-utils";
+import { PricePretty, CoinPretty, Dec, IntPretty } from "@keplr-wallet/unit";
+
+import { ObservablePool } from "@osmosis-labs/stores";
+
+import { IMPERATOR_API_DOMAIN } from "../../constants";
+import { FiatCurrency } from "@keplr-wallet/types";
+
+interface PoolMetric {
```
Comment (jonator) [inline packages/web/stores/imperator-queries/pools.ts:17]: I think you could de clutter and remove this interface and just put this type right on the `data:` field in the `ObservableImperatorQuery` base class type parameter.

**Review by jonator — CHANGES_REQUESTED** [review]
I left some comments to start. I'll address the table props problem after these changes are made. I think the table component is powerful but also not super obvious to use, sorry. Also: changes are open to discussion, please change my mind!

File: packages/web/stores/imperator-queries/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export * from "./pools";
```
[the remark this answers is not in the document]
Comment (delivan) [reply packages/web/stores/imperator-queries/index.ts:1]: First, I put it in `web` package because I thought it would only be used in `web` package and I put in `stores` folder because other query things are in `store` package.

File: packages/web/stores/imperator-queries/store.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  constructor(protected readonly kvStore: KVStore) {
+    super(() => new QueriesImperator(this.kvStore));
+  }
+
+  get(): QueriesImperator {
+    return super.get("imperator");
+  }
+}
+
+export class QueriesImperator {
```
[the remark this answers is not in the document]
Comment (delivan) [reply packages/web/stores/imperator-queries/store.ts:16]: I just intended to express it is not queries to osmosis. how about `QueriesExternal` or `ExternalQueries`?

**Comment by delivan** [comment]
Sorry for so many unused imports 😅 I made some changes. Please check it out! @jonator

File: packages/web/pages/pools/index.tsx [hunk 5]
```diff
@@ -143,63 +128,14 @@ const Pools: NextPage = observer(function () {
...
-                    0.3%
-                  </span>
-                </td>
-                <td>$466,803,653</td>
-                <td>$28,646,361</td>
-              </tr>
-            </tbody>
-          </table>
+      <section className="bg-surface shadow-separator">
+        <div className="max-w-container mx-auto p-10 py-[3.75rem]">
```
Comment (jonator) [inline packages/web/pages/pools/index.tsx:132]: Remove `mx-auto` in both sections.

**Review by jonator — CHANGES_REQUESTED** [review]
I left some more comments. Please ping me if you have questions about the `<Table />` component.

Also responded to your responses from the previous review!

File: packages/web/stores/imperator-queries/store.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (jonator) [reply packages/web/stores/imperator-queries/store.ts:16]: Yeah that makes sense. `External` works for me.

File: packages/web/hooks/use-all-pools-table.ts [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    setSortKeyPath("liquidity");
+    setSortDirection("descending");
+  }, []);
+  const baseRow: RowDef = {
+    makeHoverClass: () => "text-secondary-200",
+  };
+
+  const tableRows: RowDef[] = pools.map(() => ({
+    ...baseRow,
+    onClick: (i) => console.log(i),
```
[the remark this answers is not in the document]
Comment (delivan) [reply packages/web/hooks/use-all-pools-table.ts:135]: How about change it as a link using `next/link`, not `onClick`? It would be better for SEO and when user want to open pool detail page with a new tab.

File: packages/web/components/complex/all-pools-table-set.tsx [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+          <input
+            className="mr-2"
+            id="show-all-pools"
+            type="checkbox"
+            checked={isPoolTvlFiltered}
+          />
+          {`Show pools less then ${new PricePretty(
+            priceStore.getFiatCurrency(priceStore.defaultVsCurrency)!,
+            TVL_FILTER_THRESHOLD
+          ).toString()} TVL`}
```
[the remark this answers is not in the document]
Comment (delivan) [reply packages/web/components/complex/all-pools-table-set.tsx]: I agree with this.

File: packages/web/hooks/use-all-pools-table.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (jonator) [reply packages/web/hooks/use-all-pools-table.ts]: Ah yeah. Good idea, hadn't occurred to me.

File: packages/web/components/table/cells/pool-composition.tsx [hunk 8]
```diff
@@ -1,54 +1,64 @@
...
+export const PoolCompositionCell: FunctionComponent<PoolCompositionCell> = ({
+  value,
+  rowHovered,
+  poolId,
+}) => {
+  const { queriesOsmosisStore, chainStore } = useStore();
+
+  const queryOsmosis = queriesOsmosisStore.get(chainStore.osmosis.chainId);
+
+  const pool = queryOsmosis.queryGammPools.getPool(poolId);
```
[the remark this answers is not in the document]
Comment (jonator) [reply packages/web/components/table/cells/pool-composition.tsx:27]: My thought is it would help decouple the view from where the data comes from, even this is performant. If we refactor the structure of the queries it may be easier to continue to use this component.

TLDR: remove a dependency here.

File: packages/web/components/complex/external-incentivized-pools-table-set.tsx [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+          return pool;
+        }
+      })
+      .filter((pool: ObservablePool | undefined): pool is ObservablePool => {
+        if (!pool) {
+          return false;
+        }
+
+        const inner = ExtraGaugeInPool[pool.id];
+        const data = Array.isArray(inner) ? inner : [inner];
```
Comment (jonator) [inline packages/web/components/complex/external-incentivized-pools-table-set.tsx:45]: I just migrated this config as well for the pool detail PR. Why isn't every gauge config an array?

File: packages/stores/src/queries-external/pools.ts [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+  pool: ObservablePool;
+  liquidity: PricePretty;
+  volume24h: PricePretty;
+  fees7d: PricePretty;
+  myLiquidity?: PricePretty;
+  epochsRemaining?: number;
+  apr?: RatePretty;
+}
+
+export class ObservableExternalQuery<
```
[the remark this answers is not in the document]
Comment (jonator) [reply packages/stores/src/queries-external/pools.ts:21]: Also, could we possibly match the format of `src/queries` and put each store in its own folder? So perhaps we could make a `queries-external` folder with `pool-fees` subfolder containing the store, it's types, etc.

File: packages/web/components/control/checkbox.tsx [hunk 11]
```diff
@@ -5,13 +5,25 @@ import { Disableable, CustomClasses } from "../types";
 import { ToggleProps } from "./types";
 
 export const CheckBox: FunctionComponent<
-  ToggleProps & Disableable & CustomClasses
-> = ({ isOn, onToggle: onToggle, disabled = false, className }) => (
-  <label htmlFor="toggle-checkbox">
+  ToggleProps &
+    Disableable &
+    CustomClasses & {
+      label?: string;
```
Comment (jonator) [inline packages/web/components/control/checkbox.tsx:11]: What if we just pass the label as react children?

File: packages/web/components/table/cells/pool-composition.tsx [hunk 12]
```diff
@@ -1,54 +1,66 @@
...
+          <div className="ml-4 mr-1 flex flex-col items-start text-white-full">
+            <span
+              className={classNames({
+                "text-secondary-200": rowHovered,
+              })}
+            >
+              {pool.poolAssets.length >= 3
+                ? `${pool.poolAssets.length} Token Pool`
+                : pool.poolAssets
+                    .map((asset) => asset.amount.currency.coinDenom)
```
Comment (jonator) [inline packages/web/components/table/cells/pool-composition.tsx:43]: For some reason the denom has suddenly been displaying with the channel info.... didn't take the time to dive into the reason why.. ideas?

Screen Shot 2022-03-10 at 11 40 33 AM
Screen Shot 2022-03-10 at 11 41 05 AM

**Review by jonator — CHANGES_REQUESTED** [review]
The page is pretty demanding on my browser. May be worth profiling the components either now or once we get most of the refactor completed.

File: packages/web/components/complex/all-pools-table-set.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (victortokaito) [reply packages/web/components/complex/all-pools-table-set.tsx]: Where else do you see the use of "TVL" on the UI? 
For your side, I think it's okay to equate "TVL" with "Total Liquidity". What's easier for you to read "setIsPoolTvlFiltered" vs. "setIsPoolLiquidityFiltered"?

[the remark this answers is not in the document]
Comment (victortokaito) [reply packages/web/components/complex/all-pools-table-set.tsx]: One other reason to keep TVL would be for mobile or edge cases with limited space for text.

[the remark this answers is not in the document]
Comment (delivan) [reply packages/web/components/complex/all-pools-table-set.tsx]: @victortokaito In other components(pool card, table) the word "TVL" is not used, but "Liquidity" or "Pool Liquidity" is used. it is necessary to unify the terms not to confusing users.

File: packages/web/components/complex/external-incentivized-pools-table-set.tsx (hunk 9 above)
Comment (delivan) [reply packages/web/components/complex/external-incentivized-pools-table-set.tsx:45]: I think it should change to array

File: packages/web/components/table/cells/pool-composition.tsx (hunk 12 above)
Comment (delivan) [reply packages/web/components/table/cells/pool-composition.tsx:43]: initializing `ibcCurrencyRegistrar` is broken after `@keplr-wallet/stores` updated to 0.9.12.
It needs to contain cosmwasm query. I'm gonna fix it.

File: packages/web/components/complex/external-incentivized-pools-table-set.tsx (hunk 9 above)
Comment (jonator) [reply packages/web/components/complex/external-incentivized-pools-table-set.tsx:45]: I added this config as well for the pool detail branch (to get the bonus guages). I did the same.

File: packages/web/components/complex/all-pools-table-set.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (victortokaito) [reply packages/web/components/complex/all-pools-table-set.tsx]: Jon showed me where in the pools that was being referred to. I've updated the text to show liquidity. 

Here's an example of where we **might** have to use TVL: 
Screen Shot 2022-03-11 at 11 54 57 AM

File: packages/web/pages/pools/index.tsx [hunk 13]
```diff
@@ -37,47 +37,36 @@ const Pools: NextPage = observer(function () {
             value: <LeftTime hour="08" minute="20" />,
```
Comment (jonator) [inline packages/web/pages/pools/index.tsx:37]: Think we can get the time until epoch here?

File: packages/web/components/complex/all-pools-table-set.tsx [hunk 14]
```diff
@@ -0,0 +1,154 @@
...
+    setSortKeyPath,
+    toggleSortDirection,
+    page,
+    setPage,
+    minPage,
+    numPages,
+    tableCols,
+    tableRows,
+    tableData,
+  } = useAllPoolsTable(tvlFilteredPools, isIncentivizedPools);
```
Comment (jonator) [inline packages/web/components/complex/all-pools-table-set.tsx:97]: This may not be what you want to hear but I actually think both of these hooks should have their contents put in these `/complex/` component files. I think with the high number of output params, it's a sign that it's not worth containing within it's own componenet. If we have the code in these files with comments to describe the high level of what is happening in sections of the code then I think we've accomplished the same thing.

**Review by jonator — CHANGES_REQUESTED** [review]
We're much closer! I've made a few additional comments.

Also, think we could display a loader instead of ibc/<hash> for the table cells? See:
Screen Shot 2022-03-14 at 1 44 19 PM

File: packages/stores/src/queries-external/store.ts [hunk 15]
```diff
@@ -0,0 +1,36 @@
+import { KVStore } from "@keplr-wallet/common";
+import { HasMapStore, ObservableQuery } from "@keplr-wallet/stores";
+import { DeepReadonly } from "utility-types";
+import { ObservableQueryPoolFeesMetrics } from "./pool-fees";
+import Axios from "axios";
+
+const IMPERATOR_API_DOMAIN = "https://api-osmosis.imperator.co";
```
[the remark this answers is not in the document]
Comment (delivan) [reply packages/stores/src/queries-external/store.ts:7]: That's a good idea! 👍

File: packages/web/components/table/cells/pool-composition.tsx [hunk 16]
```diff
@@ -1,43 +1,58 @@
...
-    {poolId && tokenDenoms ? (
-      <div className="flex gap-1">
-        <div className="flex flex-col items-start text-white-full">
+  Partial<PoolCompositionCell>
+> = ({ rowHovered, poolInfoRaw }) => {
+  if (!poolInfoRaw) {
+    return null;
+  }
+
+  const poolInfo: PoolInfo = JSON.parse(poolInfoRaw!);
```
[the remark this answers is not in the document]
Comment (delivan) [reply packages/web/components/table/cells/pool-composition.tsx:32]: Oh, I misunderstood your comment 😅

**Review by jonator — CHANGES_REQUESTED** [review]
### Build Error
I'm getting the following build error:
```
./components/cards/pool-card.tsx:21:25
Type error: Type '{ amount: CoinPretty; weight: IntPretty; weightFraction: RatePretty; }[]' is not assignable to type '{ coinImageUrl: string | undefined; coinDenom: string; }[]'.
  Type '{ amount: CoinPretty; weight: IntPretty; weightFraction: RatePretty; }' is missing the following properties from type '{ coinImageUrl: string | undefined; coinDenom: string; }': coinImageUrl, coinDenom
```
### Displaying IBC assets not in assets list
Instead of showing a loader in place of `ibc/<hash>` (since this could just mean the IBC asset wasn't added to the assets list), we should instead truncate with ellipsis long denoms like this to like max 7 characters or something. So it would display ibc/9BC... instead of overflowing and causing the table to overflow. The easiest approach would be to just add a `max-w-...` to each span of the `PoolCompositionCell`.
Screen Shot 2022-03-17 at 10 50 52 AM

@victortokaito are there any more pool metrics we want to sort-by besides these? One of the main reasons for having the sort dropdown is to enable sorting by metrics that may not be table columns.
Screen Shot 2022-03-17 at 10 56 37 AM

### Performance
The initial load is slow, but perhaps we can fix perf issues from navigating back to the page (after initial load) by placing some `useMemo` hooks in the table set components. Also, I think a big source of perf issues is because for each `useFilteredData` hook we use fuse.js which stores an indexed copy of the data (pools, here). So basically the browser is dealing with 3 copies of ALL pools I believe: in the store, for the all/incentivized table, and for the external incentivized table. Perhaps we could find a way to use one `useFilteredData` hook? This may mean we should combine code even further into just the `pages/pools.tsx` component (removing the `*TableSet.tsx` components).

**Comment by jonator** [comment]
I converted this to draft because during our last frontend meeting we decided to show all superfluid pools instead of "Top Pools". Design is pending.

**Comment by delivan** [comment]
@jonator In my opinion, it is not a component issue. The main reason is that the Osmosis node is slow to respond. If you look at the Network tab on browser, you can see that the response from osmosis is very slow. 

Also, in the store, the source of pools data is only. queries in the store continue to have the first response data unless they are forced to update(because of using mobx). In component, now we just use the data. So we are not using copies of all pools. If we remove table components on pools page, we can see the perf issue is still unresolved.

I think we need a data indexing server. not query to node directly. Can we discuss about this problem with data team?

**Comment by jonator** [comment]
Also @delivan - please re-request review when you're ready for me to review.

File: packages/web/pages/pools/index.tsx [hunk 17]
```diff
@@ -34,52 +56,51 @@ const Pools: NextPage = observer(function () {
           { label: "OSMO Price", value: "$10" },
```
Comment (jonator) [inline packages/web/pages/pools/index.tsx:56]: Could we get the actual price here?

**Review by jonator — APPROVED** [review]
LGTM besides one comment. We can further polish at a later stage in the refactor.

**Review by jonator — APPROVED** [review]
